hornet-private-net/extra-nodes/private-hornet.sh

```diff
@@ -0,0 +1,323 @@
+#!/bin/bash
+
+# Script to install and deploy an additional Hornet Node to a Private Tangle
+
+# private-hornet.sh install <node_connection_str> .- Installs a new Hornet Node on your Private Tangle
+# private-hornet.sh start   <node_name> .- Starts a Hornet Node of your Private Tangle
+# private-hornet.sh update  <node_name> .- Updates a Hornet Node of your Private Tangle
+# private-hornet.sh stop    <node_name> .- Stops a Hornet Node of your Private Tangle
+
+# <node_connection_str> must be a colon-separated string in the form
+# "node_name:api_port:peering_port:dashboard_port"
+# example: "mynode:14627:15601:8082"
+# if the ports are not provided the default ones (14265, 15600, 8081) will be used
+
+# Full signature and parameters is described below: 
+# private-hornet.sh [install|update|start|stop] <node_connection_str> <coo_public_key>? <peer_multiAdress>? <snapshot_file>?
+
+
+set -e
+
+# Common utility functions
+chmod +x ../utils.sh
+source ../utils.sh
+
+help () {
+  echo "usage: private-hornet.sh [install|update|start|stop] <node_connection_str> <coo_public_key>? <peer_multiAdress>? <snapshot_file>?"
+}
+
+if [ $#  -lt 2 ]; then
+  echo "Illegal number of parameters"
+  help
+  exit 1
+fi
+
+# Prepare all execution variables
+
+command="$1"
+node_details="$2"
+
+# Split the node details string
+IFS=':'
+read -a node_params <<< "$node_details"
+
+DEFAULT_API_PORT="14265"
+DEFAULT_PEERING_PORT="15600"
+DEFAULT_DASHBOARD_PORT="8081"
+
+node_name="${node_params[0]}"
+api_port="${node_params[1]}"
+peering_port="${node_params[2]}"
+dashboard_port="${node_params[3]}"
+
+if [ -z "$api_port" ]; then
+  api_port=$DEFAULT_API_PORT
+fi
+
+if [ -z "$peering_port" ]; then
+  peering_port=$DEFAULT_PEERING_PORT
+fi
+
+if [ -z "$dashboard_port" ]; then
+  dashboard_port=$DEFAULT_DASHBOARD_PORT
+fi
+
+if [ -n "$3" ]; then
+  coo_public_key="$3"
+else 
+  if [ -f ../coo-milestones-public-key.txt ]; then
+    coo_public_key=$(cat ../coo-milestones-public-key.txt | tr -d "\n")
+  else 
+    echo "Please provide the coordinator's public key"
+    exit 131
+  fi
+fi
+
+if [ -n "$4" ]; then
+  peer_address="$4"
+else 
+  if [ -f ../node1.identity.txt ]; then
+    peer_address="/dns/node1/tcp/15600/p2p/$(getPeerID ../node1.identity.txt)"
+  else
+    echo "Please provide a peering address"
+    exit 132
+  fi
+fi
+
+if [ -n "$5" ]; then
+  snapshot_file="$5"
+else 
+  if [ -f ../snapshots/private-tangle/full_snapshot.bin ]; then
+    snapshot_file="../../../snapshots/private-tangle/full_snapshot.bin"
+  else
+    echo "Please provide a snapshot file of your Private Tangle"
+    exit 133
+  fi
+fi
+
+
+# Basic bootstrapping of folders for our Node
+
+if ! [ -d ./nodes ]; then
+  mkdir ./nodes
+fi
+
+if ! [ -d ./nodes/"$node_name" ]; then
+  mkdir ./nodes/"$node_name"
+fi
+
+cd  ./nodes/"$node_name"
+
+clean () {
+  stopContainers
+
+  if [ -d ./db ]; then
+    sudo rm -Rf ./db/*
+  fi
+
+  if [ -d ./p2pstore ]; then
```

Comment:
There is a problem here. The ./p2pstore folder is deleted right after it's created here https://github.com/iotaledger/one-click-tangle/blob/df3fd63fbb907fbf78b104552ef4fc6078c1b6a0/hornet-private-net/extra-nodes/private-hornet.sh#L143. The steps here https://github.com/iotaledger/one-click-tangle/blob/df3fd63fbb907fbf78b104552ef4fc6078c1b6a0/hornet-private-net/extra-nodes/private-hornet.sh#L180 start with the volume setup, then the clean function. The folder is created then deleted.  Then the next time the folder is created, the folder ends up in the root user group 

```
drwxrwxr-x 6 ubuntu ubuntu 4096 Aug  3 14:38 ./
drwxrwxr-x 3 ubuntu ubuntu 4096 Aug  3 14:35 ../
drwxrwxr-x 2 ubuntu ubuntu 4096 Aug  3 14:38 config/
drwxrwxr-x 2  65532  65532 4096 Aug  3 14:39 db/
-rw-rw-r-- 1 ubuntu ubuntu  702 Aug  3 14:38 docker-compose.yml
-rw-rw-r-- 1 ubuntu ubuntu  438 Aug  3 14:38 identity.txt
drwxr-xr-x 2 root   root   4096 Aug  3 14:38 p2pstore/
drwxrwxr-x 3  65532  65532 4096 Aug  3 14:35 snapshots/
```
 and we end up with this error when the node finally runs 

```The following plugins are enabled: MQTT RestAPI URTS
panic: unable to initialize data store for peer store: Error Creating Dir: "./p2pstore/node": mkdir ./p2pstore/node: permission denied

goroutine 1 [running]:
github.com/gohornet/hornet/core/p2p.provide.func1(0x0, 0x0, 0xc000010db8, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /app/core/p2p/core.go:81 +0x83d
reflect.Value.call(0x1bb4520, 0x244a198, 0x13, 0x1e43d05, 0x4, 0xc000202b40, 0x1, 0x1, 0x30, 0x7f39e7fd3108, ...)
        /usr/local/go/src/reflect/value.go:476 +0x8e7
reflect.Value.Call(0x1bb4520, 0x244a198, 0x13, 0xc000202b40, 0x1, 0x1, 0x199, 0x1, 0xc000a9f200)
        /usr/local/go/src/reflect/value.go:337 +0xb9
go.uber.org/dig.defaultInvoker(0x1bb4520, 0x244a198, 0x13, 0xc000202b40, 0x1, 0x1, 0xc000aafd10, 0xc000202b40, 0x1)
        /go/pkg/mod/go.uber.org/dig@v1.10.0/dig.go:284 +0x65
go.uber.org/dig.(*node).Call(0xc000874320, 0x2619238, 0xc000aafd10, 0x2627028, 0x1d24fa0)
        /go/pkg/mod/go.uber.org/dig@v1.10.0/dig.go:710 +0x27b
go.uber.org/dig.paramSingle.Build(0x0, 0x0, 0x0, 0x2627028, 0x1d24fa0, 0x2619238, 0xc000aafd10, 0x28, 0x1cc83c0, 0x1, ...)
        /go/pkg/mod/go.uber.org/dig@v1.10.0/param.go:245 +0x2df
go.uber.org/dig.paramObjectField.Build(...)
```

To bypass this error, I had to comment out the if statement, so that the folder isn't deleted.

Reply:
Though with this hack, if you try to recreate a node with the same name, for the node name, the p2p folder is not deleted, and a new error pops up

````
panic: unable to load/create peer identity: stored Ed25519 private key (08011240199bfbc36e18e71b1b499f899e8437e97c97d6d56b243191574170e3597a68b2c788518b31c7ae8de76feee2be6499ef714e2c9021acc0d844d2c4356b84cd47) for peer identity doesn't match private key in config (080112407ee3402bc4bc3f41868ceb3651fe0536d8740da50c56fadf96dbb69aa202ab139db4fc517f5e928855bd35c2c265926973a6a15748d149f942361d532543775a)
````

A way around this was to use a different node name in the script, which worked.

Probably a safe way to handle the ./p2p folder needs to be established.